apps/system/js/notifications.js

```diff
@@ -408,9 +408,13 @@ var NotificationScreen = {
     // timesout, so we skip updating it here, by passing a skip bool
     this.addUnreadNotification(detail.id, true);
 
+    var cIndex = this.availableNotification.indexOf(detail.appName);
+    if (cIndex === -1) {
```

Comment:
It looks like the closing brace for this if clause is 60 lines later!  And you haven't indented the intervening 60 lines correctly, so the code becomes very hard to read.

(I know I complained on an earlier version about unnecessary changes making the diff too big. But I think those were about accidental, incorrect indentation changes.  In this case, the code must be indented in order to be legible, even if it adds 60 lines to your diff.  (When something like that happens, you can always add a comment here on github letting me know that nothing has changed other than indentation.)